guardrails/guard.py

```diff
@@ -200,11 +202,16 @@ def __init__(
         self._api_client: Optional[GuardrailsApiClient] = None
         self._allow_metrics_collection: Optional[bool] = None
         self._output_formatter: Optional[BaseFormatter] = None
+        self._api_key: Optional[str] = None
+        self._base_url: Optional[str] = None
 
         # Gaurdrails As A Service Initialization
         if settings.use_server:
-            api_key = os.environ.get("GUARDRAILS_API_KEY")
-            self._api_client = GuardrailsApiClient(api_key=api_key)
+            self._api_key = api_key
```

Comment:
We should retain support for specifying these via environment variables, but prioritize them after the any values that are directly passed in.
```suggestion
            self._api_key = api_key or os.environ.get("GUARDRAILS_API_KEY")
```

Reply:
Just rechecked the defaulting in GuardrailsApiClient and what I suggest above already happens there.

Reply:
~~The GuardrailsApiClient does this for us, retrieving from environment variables if the value is `None` and also setting a default value if the environment variables aren't present. I'd prefer to have those defaults in just the one spot.~~

Slow in commenting, you saw the same thing.